Content.Server/CartridgeLoader/Cartridges/SpaceVendorsCartridgeSystem.cs

```diff
@@ -0,0 +1,70 @@
+using Content.Server.Cargo.Systems;
+using Content.Server.GameTicking;
+using Content.Server.Popups;
+using Content.Shared.CartridgeLoader.Cartridges;
+using Content.Shared.CartridgeLoader;
+using Content.Shared.IdentityManagement;
+
+namespace Content.Server.CartridgeLoader.Cartridges
+{
+    public sealed class SpaceVendorsCartridgeSystem : EntitySystem
+    {
+        [Dependency] private readonly CartridgeLoaderSystem? _cartridgeLoaderSystem = default!;
+        [Dependency] private readonly PricingSystem _pricingSystem = default!;
+        [Dependency] private readonly PopupSystem _popupSystem = default!;
+        [Dependency] private readonly GameTicker _gameTicker = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<SpaceVendorsCartridgeComponent, CartridgeUiReadyEvent>(OnUiReady);
+            SubscribeLocalEvent<SpaceVendorsCartridgeComponent, CartridgeAfterInteractEvent>(OnAfterInteract);
+        }
+
+        private void OnAfterInteract(EntityUid uid, SpaceVendorsCartridgeComponent component, CartridgeAfterInteractEvent args)
+        {
+            if (args.InteractEvent.Handled || !args.InteractEvent.CanReach || !args.InteractEvent.Target.HasValue)
+                return;
+
+            var price = _pricingSystem.GetPrice(args.InteractEvent.Target.Value);
+
+            _popupSystem.PopupEntity(Loc.GetString("price-gun-pricing-result",
+                ("object", Identity.Entity(args.InteractEvent.Target.Value, EntityManager)),
+                ("price", $"{price:F2}")), args.InteractEvent.User, args.InteractEvent.User);
+            args.InteractEvent.Handled = true;
+
+            if (component.AppraisedItems.Count >= component.MaxSavedItems)
+                component.AppraisedItems.RemoveAt(0);
```

Comment:
Should this just be a queue?

Reply:
yes, so it will be automatically sorted from the earliest scan to the oldest.